### PR TITLE
chore: Convert code to TypeScript

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

File: lib/zone.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import './browser/zone';
+
+export var Zone = global.Zone;
+
```
Comment (vicb) [inline lib/zone.ts]: I think we should apply the patches here

File: package.json [hunk 2]
```diff
@@ -33,13 +38,15 @@
     "gulp-rename": "^1.2.2",
     "gulp-uglify": "^1.2.0",
     "jasmine-core": "^2.2.0",
-    "karma": "^0.12.31",
```
Comment (vicb) [inline package.json]: remove browserify

File: test/microtasks.spec.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
 'use strict';
 
 describe('Microtasks', function () {
-  it('should execute microtasks enqueued in the root zone', function(done) {
+  it('should execute microtasks enqueued in the root global.zone', function(done) {
```
Comment (vicb) [inline test/microtasks.spec.ts]: revert

File: test/setup.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+(<any>jasmine).DEFAULT_TIMEOUT_INTERVAL = 2000;
```
Comment (vicb) [inline test/setup.ts]: could you factorize some code between microtasks / non-microtasks entry points

**Comment by vicb** [comment]
I think the bundle generation should also be updated, right ?

A few comments but the changes seem ok, good to see TS !

Before merging the PR, it would be good to test into an Ng2 app.

File: lib/zone.ts (hunk 1 above)
Comment (mhevery) [inline lib/zone.ts]: not applicable.

File: package.json (hunk 2 above)
Comment (mhevery) [inline package.json]: done

File: test/microtasks.spec.ts (hunk 3 above)
Comment (mhevery) [inline test/microtasks.spec.ts]: done

File: test/setup.ts (hunk 4 above)
Comment (mhevery) [inline test/setup.ts]: done

**Pull request closed without merging**